python/cuml/cuml/neighbors/CMakeLists.txt

```diff
@@ -34,7 +34,3 @@ rapids_cython_create_modules(
   MODULE_PREFIX neighbors_
   ASSOCIATED_TARGETS cuml
 )
-
-foreach(target IN LISTS targets_using_numpy)
-  target_include_directories(${target} PRIVATE "${Python_NumPy_INCLUDE_DIRS}")
-endforeach()
```

Comment:
Trying dropping this. AFAICT the Cython modules above don't `cimport numpy`. So they wouldn't need this

Not sure whether it would cause the tests to hang. At a minimum, it is unused; so, it is worth cleaning up

Reply:
Yeah not sure what happened there. I traced back in the blame to where this was added. Looks like @vyasr recommended removing it back at the source: https://github.com/rapidsai/cuml/pull/4818#discussion_r932650099

But there wasn't any additional discussion on that PR (maybe it happened somewhere else), and the change was merged in.

I agree with you that it seems to be unused.

Reply:
Originally was looking for clues on fixing the hanging test ( https://github.com/rapidsai/cuml/pull/6031#issuecomment-2313885246 ). Tried this just in case it helped, but it didn't matter

Read through the history here yesterday. It seems like NumPy was a build dependency a while back (though still wasn't clear then whether it was being used). Think since then every update has assumed NumPy was a build dependency. However as we don't require it during the build, it isn't actually satisfied.

Further we would have needed `find_package(Python REQUIRED COMPONENTS Development NumPy)` to find NumPy and set `$Python_NumPy_INCLUDE_DIRS`, but we don't do that either.

Think this hasn't presented much of an issue as we don't actually set `targets_using_numpy`.

In any event, this time seems as good as any for cleaning this up